metricflow/dataset/sql_dataset.py

```diff
@@ -181,18 +183,30 @@ def instance_for_column_name(self, column_name: str) -> MdoInstance:
         )
 
     def instance_from_time_dimension_grain_and_date_part(
-        self, time_dimension_spec: TimeDimensionSpec
+        self, time_granularity_name: Optional[str] = None, date_part: Optional[DatePart] = None
     ) -> TimeDimensionInstance:
-        """Find instance in dataset that matches the grain and date part of the given time dimension spec."""
+        """Find instance in dataset that matches the given grain and date part."""
         for time_dimension_instance in self.instance_set.time_dimension_instances:
             if (
-                time_dimension_instance.spec.time_granularity == time_dimension_spec.time_granularity
-                and time_dimension_instance.spec.date_part == time_dimension_spec.date_part
+                time_dimension_instance.spec.time_granularity_name == time_granularity_name
+                and time_dimension_instance.spec.date_part == date_part
+                and not time_dimension_instance.spec.window_functions
             ):
                 return time_dimension_instance
 
         raise RuntimeError(
-            f"Did not find a time dimension instance with matching grain and date part for spec: {time_dimension_spec}\n"
+            f"Did not find a time dimension instance with grain '{time_granularity_name}' and date part {date_part}\n"
```

Comment:
This error would be easier to read via `str(LazyFormat(...))` or `mf_pformat_dict(...)`. Same for other cases.